### PR TITLE
Add some tests for getHeads

### PR DESCRIPTION
**Pull request by Powersource**
Helps with solving https://github.com/ssbc/ssb-tangle/issues/7

### CODE REVIEW

File: test/lib/misc-junk.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  t.deepEqual(getHeads(inputMap), expectedHeads, 'one head')
+
+  t.end()
+})
+
+test('getHeads: Root-only graph', t => {
+  //    A   (root)
+
+  const inputMap = {
+    A: {}
```
Comment (Powersource) [inline test/lib/misc-junk.test.js:24]: Map() never outputs this atm but it's my proposal for how it would look after https://github.com/ssbc/ssb-tangle/issues/5

**Comment by Powersource** [comment]
For a lack of CI, the failing tests are
```
getHeads: Root-only graph
getHeads: Graph with simple merge
```

**Comment by Powersource** [comment]
wait I remembered another case I want to add

File: test/lib/misc-junk.test.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  //          ----?--- J? (a message we don't have)
+  //                   |
+  //                   K
+
+  const inputMap = {
+    A: {},
+    J: { K: 1 }
+  }
+
+  const expectedHeads = ['K']
```
Comment (Powersource) [inline test/lib/misc-junk.test.js:46]: Ok had to add this one because it was fucking with my head. I think a naive (or maybe reasonable) algorithm would respond `['A', 'K']` since they look like heads at first glance. But I think `['K']` is technically correct since `A` is the root and so we know that `K` is a descendant of it (i.e. `A` is not a head).

Also hehe I think accidentally this test is passing, but I think that's for the same reason that `getHeads: Root-only graph` is failing: the root is ignored.

**Comment by Powersource** [comment]
Ok *now* I'm done

File: test/lib/misc-junk.test.js (hunk 2 above)
Comment (Powersource) [reply test/lib/misc-junk.test.js:46]: Ok there's no way for getHeads to know that A is the root, plus if this map/head logic should work for sub(partial?) graphs then it also sort of makes sense to have both as heads.

So I think I think this line should be changed to `['A', 'K']`

**Comment by mixmix** [comment]
thanks so much for adding these tests @Powersource ! Sorry for the long delay. Please feel free to DM me on ssb in general (I honestly don't watch notifications anywhere else).

I notice this method isn't used anywhere in the repo currently, and I think in fact I've been using the `reduce` function to do this. But I have an emerging need to just calculate heads without reducing, so will see what I can do about getting a working function.

Basically I think the API needs to include the entry point

**Comment by mixmix** [comment]
Also, I think the file-name `lib/misc-junk.js` was quite deliberately dodgy sounding. i.e. if you're using things from here ... look out!

**Comment by mixmix** [comment]
Hey @Powersource I've written a function for finding heads in a way which I think is best.
The tests should show how it can be used, let me know if it's not clear though.

NOTE - this ignores any dangling nodes, because for my needs and assumptions about tangles I'm working with, that's appropriate. If you want to include dangling nodes, perhaps check out `ssb-sort` which is the old (and rough)  tangle heads calculator.

**Comment by mixmix** [comment]
published as `ssb-tangle@2.1.0`
